lib/board.rb

```diff
@@ -0,0 +1,92 @@
+class Board
+
+    def initialize()
+        @grid = Array.new(3){Array.new(3, '_')}
+    end
+
+    def get_grid
+        @grid
```

Comment:
It can be helpful to return what's called a "defensive copy" of `@grid` here, so that code on the outside of `Board` can't modify the Board's internal data.  By that I mean it might be possible to do the following:

```ruby
board = Board.new
grid = board.get_grid
grid[0] = "not an array"

board.make_move(0, 0, "X") #this code would fail now
```

Reply:
This can be avoided by returning a copy of the data instead, if it is being returned outside the class.

```ruby
class Board
  def get_grid
    grid_copy = @grid.map do |row|
      row_copy = row.map { |cell| cell }
      row_copy
    end
  end
end

board = Board.new
grid = board.get_grid
grid[0] = "not an array"

board.make_move(0, 0, "X") #now this works because Board.@grid hasn't changed
```